### PR TITLE
Project Weather App - Vanessa Sue Smith

### PR DESCRIPTION
**Pull request by VanessaSue27**
Link to my deployed project:
https://frosty-murdock-a67563.netlify.app/

My reflections on what new things I learned and how I approached building this project are noted down on the Read Me file.

I have some additional observations:
- I chose to use "Max temperature" and "Feels Like temperature" for my Forecast section, since Min and Max temperature in this API were always returning values too similar to each other, so it felt redundant to list to equal numbers next to each other.

- One small bug I noticed on the API's 5 day Forecast endpoint: It doesn't generate the data starting tomorrow, it will generate the data starting today and then the coming 4 days.
I tested it several times and this seems to be the case when I check the app some time before 12:00. If I check the app again in the evening, the data will start effectively from the day after. So the result fluctuates (probably something to do with our filter being set at 12:00). Asked Maks about this and he attributed it to poorly developed API, so we have this limitation.

- I set everything up for key.js being in a hidden file by including it in .gitignore, but I had to make it public again in order for Netlify to copy the file and be able to complete the fetch URL.

- My goal for this project was to spend more time on the design, which I felt I had neglected a bit in the last project. Since the HTML elements are dynamically created and not hardcoded, it was challenging creating the Flexboxes I needed and having a good overview of what it is exactly that I am styling; but I am proud of the result and that I managed different styles for the different devices.

### CODE REVIEW

File: code/index.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Project: Weather App</title>
+    <link rel="stylesheet" href="./style.css">
+    <link href="https://fonts.googleapis.com/css2?family=Kumbh+Sans:wght@400;700&display=swap" rel="stylesheet">
+</head>
```
Comment (ckjerrgren) [inline code/index.html:9]: A little favicon would be nice for your next project :)

**Review by ckjerrgren — COMMENTED** [review]
Vanessa it is a beautiful website. I have no criticism. 
The code was clean and easy to follow, I love that you challenged yourself to spend more time on styling this week and it looks fab. 
Great job with the meta. 

Overall a job well done. Hopefully the teachers have something more constructive to say. :)

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+.DS_Store
+.vscode
```
Comment (EmOlsz) [inline .gitignore:2]: Very good Vanessa!

File: code/key.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export const API_KEY = '3f1c95b540a45d2a48ff596267d9d939';
```
Comment (EmOlsz) [inline code/key.js:1]: 👍

File: code/script.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+const sunrise = document.getElementById('sunrise');
+const sunset = document.getElementById('sunset');
+
+//Main fetch to get Stockholm weather
+fetch(`https://api.openweathermap.org/data/2.5/weather?q=Stockholm&units=metric&APPID=${API_KEY}`)
+  .then((response) => {
+    return response.json();
+  })
+
```
Comment (EmOlsz) [inline code/script.js:16]: No need for this blank line, it should be deleted

File: code/script.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  .then((json) => {
```
Comment (EmOlsz) [inline code/script.js:17]: since it's one argument, we can omit extra pair of parentheses

File: code/script.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    const set = new Date(sunsetValue * 1000);
+    const sunriseHour = sun.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit', hour12: false,});
+    const sunsetHour = set.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit', hour12: false,});
+
+    sunrise.innerText = `Sunrise: ${sunriseHour}`;
+    sunset.innerText = `Sunset: ${sunsetHour}`;
+  })
+
+  .catch((error) => {
+    console.log(error);
```
Comment (EmOlsz) [inline code/script.js:39]: there is a batter type of `console` method for that  - `console.error()`. Thanks to that, it's going to have red background

File: code/style.css [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+
+html, body {
+    height: 100%;
+    position: relative;
+}
+
+body {
+    margin: 0;
+    font-family: 'Kumbh Sans', sans-serif;
+    background-image: url(./assets/background-img.png);
```
Comment (EmOlsz) [inline code/style.css:19]: it's better to wrap content of url in quotes - some browsers might have problem with reading it without

```
background-image: url("./assets/background-img.png");
```

File: code/style.css [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+
+/*Styling Main Top section */
+.icon-temp {
+    display: flex;
+    width: 100%;
+    justify-content: space-around;
+    align-items: flex-end;
+}
+
+#temperature {
```
Comment (EmOlsz) [inline code/style.css:39]: Styling by id - big no-no

**Review by EmOlsz — APPROVED** [review]
Good job Vanessa! Apart from basic requirements, you've also added some extra stuff, appreciate it :) Your code is well done and structured. Extra congrats for mobile first.
There are some things you could improve - please, take a look on my comments and remember about them for future projects

**Pull request closed without merging**